react/features/base/conference/middleware.any.ts

```diff
@@ -195,64 +200,72 @@ function _conferenceFailed({ dispatch, getState }: IStore, next: Function, actio
-                    .then(conference.leave())
-                    .then(dispatch(disconnect()))
-                    .then(dispatch(connect()));
-            }
+        dispatch(showErrorNotification({
+            hideErrorSupportLink: true,
+            descriptionKey: 'dialog.maxUsersLimitReached',
+            titleKey: 'dialog.maxUsersLimitReachedTitle'
+        }, NOTIFICATION_TIMEOUT_TYPE.LONG));
+
+        // in case of max users(it can be from a visitor node), let's restore
+        // oldConfig if any as we will be back to the main prosody
+        const newConfig = restoreConferenceOptions(getState);
+
+        if (newConfig) {
+            dispatch(overwriteConfig(newConfig)) // @ts-ignore
+                .then(dispatch(conferenceWillLeave(conference)))
+                .then(conference.leave())
+                .then(dispatch(disconnect()))
+                .then(dispatch(connect()));
         }
 
         break;
     }
     case JitsiConferenceErrors.OFFER_ANSWER_FAILED:
         sendAnalytics(createOfferAnswerFailedEvent());
         break;
-    case JitsiConferenceErrors.REDIRECTED: {
-        // once conference.js is gone this can be removed and both
-        // redirect logics to be merged
-        if (typeof APP === 'undefined') {
-            const newConfig = getVisitorOptions(getState, error.params);
-
-            if (!newConfig) {
-                logger.warn('Not redirected missing params');
-                break;
-            }
-
-            const [ vnode ] = error.params;
 
-            dispatch(overwriteConfig(newConfig)) // @ts-ignore
-                .then(dispatch(conferenceWillLeave(conference)))
-                .then(conference.leave())
-                .then(dispatch(disconnect()))
-                .then(dispatch(setIAmVisitor(Boolean(vnode))))
+    case JitsiConferenceErrors.REDIRECTED: {
+        const newConfig = getVisitorOptions(getState, error.params);
 
-                // we do not clear local tracks on error, so we need to manually clear them
-                .then(dispatch(destroyLocalTracks()))
-                .then(dispatch(connect()));
+        if (!newConfig) {
+            logger.warn('Not redirected missing params');
+            break;
         }
+
+        const [ vnode ] = error.params;
+
+        dispatch(overwriteConfig(newConfig)) // @ts-ignore
+            .then(dispatch(conferenceWillLeave(conference)))
+            .then(conference.leave())
+            .then(dispatch(disconnect()))
+            .then(dispatch(setIAmVisitor(Boolean(vnode))))
+
+            // we do not clear local tracks on error, so we need to manually clear them
+            .then(dispatch(destroyLocalTracks()))
+            .then(() => {
+                if (typeof APP !== 'undefined') {
+                    // Reset VideoLayout. It's destroyed in features/video-layout/middleware.web.js so re-initialize it.
+                    VideoLayout.initLargeVideo();
```

Comment:
This is not going to work, as I mentioned above. Maybe we can introduce a new action _CONFERENCE_WILL_REDIRECT and re-create the video layout there, in a web-only middleware.